### PR TITLE
fix(mutable_buffer): Statistics incorrect when subset of columns is inserted

### PR DESCRIPTION
**Pull request by alamb**
Part of https://github.com/influxdata/influxdb_iox/issues/2124

# Rationale
When inserting batches into the MUB that do have all columns, the values were padded out to the correct row counts with nulls, but the statistics were not.

# Changes
1. Pad statisics as well
2. Tests for same

### CODE REVIEW

File: mutable_buffer/src/chunk.rs [hunk 1]
```diff
@@ -502,6 +503,86 @@ mod tests {
...
+        let lp4 = "cpu new_field=200 4";
+
+        let mut chunk = write_lp_to_new_chunk(lp1).unwrap();
+        write_lp_to_chunk(&lp2, &mut chunk).unwrap();
+        write_lp_to_chunk(&lp3, &mut chunk).unwrap();
+        write_lp_to_chunk(&lp4, &mut chunk).unwrap();
+
+        let summary = chunk.table_summary();
+
+        let expected_column_summaries = vec![
```
Comment (alamb) [inline mutable_buffer/src/chunk.rs:530]: There were some great values in here originally (like total_count = 3 for some columns and total_count = 4 for others)

Comment (NGA-TRAN) [reply mutable_buffer/src/chunk.rs:530]: Nice to see all columns in same chunk have same count.

File: mutable_buffer/src/column.rs [hunk 2]
```diff
@@ -266,12 +268,30 @@ impl Column {
...
+                stats.update_for_nulls(delta as u64);
+            }
+            ColumnData::Bool(data, stats) => {
+                data.append_unset(delta);
+                stats.update_for_nulls(delta as u64);
+            }
+            ColumnData::Tag(data, _dict, stats) => {
+                data.resize(len, INVALID_DID);
+                stats.update_for_nulls(delta as u64);
+            }
```
Comment (NGA-TRAN) [inline mutable_buffer/src/column.rs:294]: nice

**Review by NGA-TRAN — APPROVED** [review]
LGTM

File: mutable_buffer/src/column.rs (hunk 2 above)
Comment (alamb) [reply mutable_buffer/src/column.rs:294]: 😆  yes that seems like an improvement for sure

**Pull request merged**